src/lib.rs

```diff
@@ -456,21 +489,68 @@ where
     }
 }
 
-// To allow the module submitting unsigned transaction
-impl<T: Trait> frame_support::unsigned::ValidateUnsigned for Module<T> {
-    type Call = Call<T>;
-
-    fn validate_unsigned(_source: TransactionSource, call: &Self::Call) -> TransactionValidity {
-        if let Call::clear_ocwtasks() = call {
-            // TODO: validate the signed payload and verify the payload here
-            ValidTransaction::with_tag_prefix("product-tracking-ocw")
-                .priority(100)
-                .and_provides([b"clear_ocwtasks"])
-                .longevity(3)
-                .propagate(true)
-                .build()
-        } else {
-            InvalidTransaction::Call.into()
+pub struct ShippingEventBuilder<Moment>
```

Comment:
I am thinking if we can put `ShippingEvent` and `ShippingEventBuilder` into a separate file/mod